diff --git a/kernel/exit.c b/kernel/exit.c
index 54c3269..68f1967 100644
--- a/kernel/exit.c
+++ b/kernel/exit.c
@@ -767,11 +767,24 @@ static void check_stack_usage(void)
 static inline void check_stack_usage(void) {}
 #endif
 
+static unsigned int global_func_exe_times = 0;
+
+void trace_function_dummy_call(void)
+{
+	trace_printk("[++] comm:%s, pid:%d, from(%pS) \n",
+			current->comm, current->pid, (void *)__builtin_return_address(0));
+}
+
 void __noreturn do_exit(long code)
 {
 	struct task_struct *tsk = current;
 	int group_dead;
 
+	trace_printk("[+] comm: %s, pid: %d, global_func_exe_times: %d\n",
+			current->comm, current->pid, global_func_exe_times);
+	global_func_exe_times++;
+	trace_function_dummy_call();
+
 	profile_task_exit(tsk);
 	kcov_task_exit(tsk);
 
@@ -815,6 +828,9 @@ void __noreturn do_exit(long code)
 		schedule();
 	}
 
+	trace_printk("[-] comm: %s, pid: %d, global_func_exe_times: %d\n",
+			current->comm, current->pid, global_func_exe_times);
+
 	exit_signals(tsk);  /* sets PF_EXITING */
 	/*
 	 * Ensure that all new tsk->pi_lock acquisitions must observe